modules/ext.matomoanalytics.graphs.js

```
@@ -0,0 +1,115 @@
+// eslint-disable-next-line no-undef
+Chart.register({
+	id: 'noData',
+	afterDraw: function (chart) {
+		var datasets = chart.data.datasets;
+
+		// Improved data check to exclude NaN, null, undefined, or 0 values
+		var hasData = datasets.some(dataset => 
+			dataset.data.length > 0 && 
+			dataset.data.some(value => value !== null && value !== undefined && value !== 0 && !isNaN(value))
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)